stdlib/public/stubs/Stubs.cpp

```diff
@@ -316,6 +343,34 @@ __mulodi4(di_int a, di_int b, int* overflow)
 }
 #endif
 
+#if defined(__CYGWIN__)
+// Cygwin does not have strtoXX_l() functions.
+// These are wrapper functions only for C locale.
+static long double strtold_l(const char *nptr, char **EndPtr, locale_t) {
+  std::string lastLocale = setlocale(LC_ALL, "");
+  setlocale(LC_ALL, "C");
```

Comment:
Isn't this setlocale call affecting the whole app?  Possibly changing the behavior for other threads?